### PR TITLE
chore: run go-sdk generation during maven build

### PR DESCRIPTION
**Pull request by jsenko**

### CODE REVIEW

File: go-sdk/generate-docker.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+then
+  echo "Error: 'docker' executable not found"
+  exit 1
+fi
+
+read -r -d '' SCRIPT <<- END
+apt-get update &&
+apt-get -y install unzip &&
+apt-get -y install libicu-dev &&
+chmod u+rw /registry &&
```
Comment (andreaTP) [inline go-sdk/generate-docker.sh:15]: This reminds me that the generated files are going to belong to the user used in the container(`root`?) ...

File: go-sdk/generate-docker.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/registry/go-sdk/generate.sh &&
```
Comment (andreaTP) [inline go-sdk/generate-docker.sh:16]: if you `cd` into the go-sdk folder and run:

```
make generate
make format
```
should be enough

File: go-sdk/generate.sh [hunk 3]
```diff
@@ -39,8 +39,8 @@ cp $SCRIPT_DIR/../common/src/main/resources/META-INF/openapi-v2.json $SCRIPT_DIR
 cp $SCRIPT_DIR/../common/src/main/resources/META-INF/openapi.json $SCRIPT_DIR/v3.json
 
 # Hask to overcome https://github.com/microsoft/kiota/issues/3920
-$SED_NAME -i 's/NewComment/DTONewComment/' v2.json
-$SED_NAME -i 's/NewComment/DTONewComment/' v3.json
+$SED_NAME -i 's/NewComment/DTONewComment/' $SCRIPT_DIR/v2.json
```
Comment (andreaTP) [inline go-sdk/generate.sh:42]: Good one!
I think that this has been fixed in upstream Kiota, it might be time to remove this hack

**Review by andreaTP — CHANGES_REQUESTED** [review]
I'm mostly worried about ownership and permissions of the generated files across machines/developers.
Happy to learn that everything will be fine though.

File: go-sdk/generate.sh (hunk 3 above)
Comment (jsenko) [reply go-sdk/generate.sh:42]: I'm still getting
```
# github.com/apicurio/apicurio-registry/go-sdk/pkg/registryclient-v3/models
pkg/registryclient-v3/models/new_comment.go:8:6: NewComment redeclared in this block
	pkg/registryclient-v3/models/comment.go:23:6: other declaration of NewComment
```
when running `make test`.

**Comment by jsenko** [comment]
> I'm mostly worried about ownership and permissions of the generated files across machines/developers. Happy to learn that everything will be fine though.

The permissions are fine on my machine:
Files `-rw-r--r--.`
Dirs `drwxr-xr-x.`

File: go-sdk/generate-docker.sh (hunk 1 above)
Comment (jsenko) [reply go-sdk/generate-docker.sh:15]: I've tested this and the files belong to the user who is running the script (from the outside).

**Comment by jsenko** [comment]
hi @andreaTP What do you suggest to do with python? I've noticed that we do not commit the python generated files, so is a script needed? 
Anyway I tested running `poetry install` in `python:latest` and got an error:
```
Preparing build environment with build-system requirements poetry-core, requests, shutils
Traceback (most recent call last):
  File "/python-sdk/kiota-gen.py", line 88, in <module>
    generate_kiota_client_files({})
  File "/python-sdk/kiota-gen.py", line 27, in generate_kiota_client_files
    if platform.processor()[0] == "arm" and kiota_arch_name == "x64":
       ~~~~~~~~~~~~~~~~~~~~^^^
IndexError: string index out of range

Command ['/tmp/tmpzhxi09u3/.venv/bin/python', '/python-sdk/kiota-gen.py'] errored with the following return code 1
```
Command history:
```
cd /python-sdk
apt update
apt install pipx
pipx ensurepath
export PATH="$PATH:/root/.local/bin"
pipx install poetry
make install
```

File: go-sdk/generate.sh (hunk 3 above)
Comment (andreaTP) [reply go-sdk/generate.sh:42]: ok, let's test the fix next time 👍

**Comment by andreaTP** [comment]
> What do you suggest to do with python? I've noticed that we do not commit the python generated files, so is a script needed?

Thanks for the question! Thinking twice we don't probably need to do anything, as the sources are not committed as you noticed ...

> platform.processor()

This seems to be a bug: https://stackoverflow.com/a/63755732 should be straightforward to fix but we might need @carlesarnal to validate the fix as well ... 😅

**Comment by andreaTP** [comment]
Thanks @jsenko ! 🎉

File: go-sdk/generate-docker.sh (hunk 1 above)
Comment (EricWittmann) [reply go-sdk/generate-docker.sh:15]: For whatever reason this definitely not true for me.  Generated files are created as root for me.  Might depend on how you have docker configured, or if you're using docker vs. podman.

**Pull request merged**